### ISSUE TITLE
Misleading (?) wording in documentation

### ISSUE THREAD

**Issue by brunoscopelliti** [facebook/react#17090]
> A custom Hook is a JavaScript function whose name starts with ”use” and that **may** call other Hooks.

Source: https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook

What I find misleading is the verb _may_.

What's the point of a custom hook that doesn't call any React's hook internally? It doesn't hook to React's internals... it also needn't to follow the rules of hook, and still works perfectly fine. I would say it is just a normal function whose name starts with _use_. 

Am I missing something?

**Comment by wendy81** [comment]
yes, you can change your normal function whose name starts with use, and it works well too. It's only a normal function name, that's all.
If you don't use any React's hook in normal function, why not use a normal function name? 
That's we could say: What's the point of a custom hook that doesn't call any React's hook internally?

But if you want to add any React's hook in a normal function, you have to named its name starts with use, otherwise  you will trigger error information, as below error:

// Normal function below
function test() {  // this name doesn't start with use.
    const [counter] = useState(0);   //  use hook in the normal function 
    return <div>{counter}</div>;
}
so you will get error information .

" error  React Hook "useState" is called in function "test" which is neither a React function component or a custom React Hook function  react-hooks/rules-of-hooks".

This is my understanding,  hope to be useful for you.

**Comment by brunoscopelliti** [comment]
@wendy81 I am not sure I get the point of your comment.

**Comment by wendy81** [comment]
> @wendy81 I am not sure I get the point of your comment.

two points:
1.  a normal function whose name starts with use, that's ok. It's just a normal function.
2. but if this normal function you add any React's hook, this function's name must start with use. Otherwise, it'll get a error.

Is it clear now?

**Comment by brunoscopelliti** [comment]
Yes, that's clear... but at this point I'm not sure my question was clear.
Let me rephrase it.

React docs states that a custom hook is a JavaScript function whose name starts with ”use” and that may call other Hooks.
So a custom hook might as well not call any other hooks. This part is a bit confusing to me, because as you said this wouldn't be an hook at all, just an ordinary function that happen to have a name starting with the _use_ prefix.

As definition for a custom hook I would have expected something like:

> A custom Hook is a JavaScript function whose name starts with ”use” and that call **one, or more** other Hooks.

Unless I'm missing something, and in this case please let me know what.

**Comment by Shivansh2407** [comment]
Please can I help on this issue?

**Comment by vkurchatkin** [comment]
> So a custom hook might as well not call any other hooks. 

Yes, that is true, and it will still be a hook. When you say that a function is a hook, it means that the user of this function must follow the rules of hooks, no matter what the implementation currently is. 

For example, you can do this:

```js
function useCurrentUser() {
  return {
     username: 'Foo'
  };
}
```

And then swap it for the actual implementation without breaking possible callers:

```js
function useCurrentUser() {
  return useContext(UserContext);
}
```

**Comment by brunoscopelliti** [comment]
Ok, i see. Thank you.
